alleycats-core/src/main/scala/alleycats/std/set.scala

```diff
@@ -68,6 +68,9 @@ object SetInstances {
       def foldRight[A, B](fa: Set[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
         Foldable.iterateRight(fa, lb)(f)
 
+      override def foldMap[A, B](fa: Set[A])(f: A => B)(implicit B: Monoid[B]): B =
+        B.combineAll(fa.map(f))
```

Comment:
`fa.iterator.map(f)` will be faster (not allocate a set) also maybe more correct. `fa.map(f)` if `f = { _ => 1 }` will create a set of 1 thing. If you do `foldMap` what do we mean? I actually don't know if this should return the count of items (which I think it should, which means we should do `fa.iterator.map`), or if it should return 0 if empty and 1 otherwise.

I guess since this is lawless especially we need a comment as to the behavior.

Reply:
That's a good point, I originally avoided using `.iterator` to maintain the same semantics, but I forgot to confirm that was the case.

```scala
@ alleycats.std.SetInstances.setTraverse.foldMap(Set(1,2,3))(_ => "a")
res13: String = "aaa"
```

Seems _not_ using `.iterator` actually would be a breaking change, so I'll switch to use it. Nice catch!